pht_federated/aggregator/api/endpoints/discovery.py

```diff
@@ -0,0 +1,433 @@
+                fig_json = plotly.io.to_json(figure)
+                obj = json.loads(fig_json)
+
+                figure_schema = {
+                    'title': discovery_title,
+                    'type': "numeric",
+                    'figure': obj
+                }
+
+                discovery_figure = DiscoveryFigure(**figure_schema)
+                discovery_summary_json['figure_data'] = discovery_figure
+                aggregated_feature_lst.append(discovery_summary_json)
+
+                if len(feature_lst) == 0:
+                    break
+
+            elif feature['type'] == 'unstructured':
+
+                discovery_summary_json = {
+                    "type": 'unstructured'
+                }
+                aggregated_feature_lst.append(discovery_summary_json)
+
+            elif feature['type'] == 'unique':
+
+                value = feature
+                discovery_title = ""
+                discovery_no_duplicates = 0
+
+                for feature2 in feature_lst:
+                    if feature2['title'] == value['title']:
+                        data = feature2
+                        discovery_title = data['title']
+                        discovery_no_duplicates += data['number_of_duplicates']
+
+                feature_lst = [x for x in feature_lst if x['title'] != discovery_title]
+                if len(feature_lst) == 0:
+                    break
+
+                discovery_no_duplicates /= len(response)
+
+                discovery_summary_json = {
+                    "type": 'unique',
+                    "title": discovery_title,
+                    "number_of_duplicates": discovery_no_duplicates
+                }
+                aggregated_feature_lst.append(discovery_summary_json)
+
+            elif feature['type'] == 'equal':
+
+                value = feature
+                discovery_title = ""
+
+                for feature2 in feature_lst:
+                    if feature2['title'] == value['title']:
+                        data = feature2
+                        discovery_title = data['title']
+                        discovery_equal_value = data['value']
+
+                feature_lst = [x for x in feature_lst if x['title'] != discovery_title]
+                if len(feature_lst) == 0:
+                    break
+
+                discovery_summary_json = {
+                    "type": 'equal',
+                    "title": discovery_title,
+                    "value": discovery_equal_value
+                }
+                aggregated_feature_lst.append(discovery_summary_json)
+
+
+
+        discovery_feature_count /= len(response)
+
+        discovery_summary_schema = {
+            "proposal_id": proposal_id,
+            "item_count": discovery_item_count,
+            "feature_count": discovery_feature_count,
+            "data_information": aggregated_feature_lst
+        }
+
+        discovery_summary = DiscoverySummary(**discovery_summary_schema)
```

Comment:
split into subfunctions and move to different file. function should have ~10 lines